toolkits/landcover/impl/landsat8.js

```diff
@@ -41,16 +41,19 @@ var QA_BAND = 'pixel_qa';
  * Returns a new Landsat 8 SR dataset instance.
  *
  * @constructor
+ * @param {ee.ImageCollection=} collection The Landsat image collection backing
```

Comment:
It seems kind of confusing to allow passing in a landsat collection to the thing that's supposed to generate it.

Reply:
This allows us to test the interface via the "front door" rather than by poking holes in the loosey JavaScript classes, and it allows the client to pass their own collection in case they already have one half built. I can remove this in the new PR.